CMakeLists.txt

```diff
@@ -1,10 +1,72 @@
 # Native code side of Thrive
-cmake_minimum_required(VERSION 3.10)
+cmake_minimum_required(VERSION 3.13)
 
-project(Thrive)
+# Must be defined before project() call
+set(CMAKE_CONFIGURATION_TYPES "Debug;Release;RelWithDebInfo;Distribution" CACHE STRING "" FORCE)
 
-# If you want to get compile commands run cmake with
-# "-DCMAKE_EXPORT_COMPILE_COMMANDS=ON"
+project(Thrive VERSION 1.0 LANGUAGES CXX)
+
+# Add our custom CMake scripts directory
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/Scripts")
+include(CMakeConfiguration)
+include(CMakeHelperFunctions)
+
+if(WIN32)
+    set(CMAKE_GENERATOR_PLATFORM x64)
+endif()
+
+message(STATUS "Building for ${THRIVE_OS} on ${THRIVE_ARCH}")
+
+# Detect library version
```

Comment:
I think this code still very much needs to be after the user-configurable option variables which I think should be as close to the top of the file as possible. I'd put them even way before including any of the helper functions etc.